### PR TITLE
Create endpoints and query functions for defined and custom period

### PR DESCRIPTION
**Pull request by Antyfreezze**

### CODE REVIEW

File: database/helpers.py [hunk 1]
```diff
@@ -1,11 +1,18 @@
 import psycopg2
+from datetime import datetime, timedelta
 from aiopg.sa import create_engine
 from config import db
 from sqlalchemy.schema import CreateTable, DropTable
+from sqlalchemy import select, and_
 from database.models import Transaction
+from database.models import Category
+from sqlalchemy import create_engine as alchemy_create_engine
```
Comment (sapphireua) [inline database/helpers.py]: Please sort imports due to https://www.python.org/dev/peps/pep-0328/ . If you use Pycharm, you can execute Optimize imports command (https://www.jetbrains.com/help/pycharm/creating-and-optimizing-imports.html)

File: database/helpers.py [hunk 2]
```diff
@@ -88,3 +95,62 @@ def _convert_resultproxy_to_dictionary(result_proxy):
     for row in result_proxy:
         dict_result.append(dict(row))
     return dict_result
+
+
+async def get_data_define_period(category_name, period):
```
Comment (sapphireua) [inline database/helpers.py]: Please add typing

File: database/helpers.py [hunk 3]
```diff
@@ -88,3 +95,62 @@ def _convert_resultproxy_to_dictionary(result_proxy):
...
+async def get_data_custom_period(category_name, start_date, end_date):
+    try:
+        s_date = datetime.strptime(start_date, "%Y-%m-%d")
+        e_date = datetime.strptime(end_date, "%Y-%m-%d")
+        if s_date > e_date:
+            start_date, end_date = end_date, start_date
+    except ValueError:
+        return "Invalid dates"
+
+    engine = alchemy_create_engine(dsn)
```
Comment (sapphireua) [inline database/helpers.py]: See comment related to engine above

File: database/helpers.py [hunk 4]
```diff
@@ -88,3 +95,62 @@ def _convert_resultproxy_to_dictionary(result_proxy):
...
+    conn = engine.connect()
```
Comment (sapphireua) [inline database/helpers.py]: From line `106` to `126` you have almost the same code as `139` to `156`. You can split some gode to functions and reuse it.

File: database/helpers.py [hunk 5]
```diff
@@ -88,3 +90,53 @@ def _convert_resultproxy_to_dictionary(result_proxy):
...
+
+
+def _date_validator(period=None, start_date=None, end_date=None) -> tuple:
+    """ Function for syncing to database format """
+    if period:
+        end_date = str(datetime.date(datetime.now()))
+        start_date = str(datetime.date(datetime.now()) - timedelta(int(period)))
+    else:
+        try:
+            s_date = datetime.strptime(start_date, "%Y-%m-%d")
```
Comment (Dmitriy00047) [inline database/helpers.py]: Do we use format like this?

File: monefystat_api/api_v1.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 from sanic import Blueprint
 from monefystat_api.service_resource import drop_endpoint, data_endpoint
 from monefystat_api.service_resource import smoke_endpoint, webhook_enable, webhook_reciver, create_endpoint
-
+from monefystat_api.service_resource import get_data_for_custom_period_endpoint, get_data_for_defined_period_endpoint
```
Comment (Dmitriy00047) [inline monefystat_api/api_v1.py]: merge imports

**Pull request merged**